sampuru-server/src/main/java/ca/on/oicr/gsi/sampuru/server/DBConnector.java

```diff
@@ -100,83 +102,81 @@ public void writeDeliverables(JSONArray deliverableArray, String username) throw
             }
         }
 
-        //TODO: Adding Collection of VALUEs to Insert is coming in jOOQ 3.15 apparently. They disapprove of this approach
         if(!unknownDeliverables.isEmpty()) {
             try(final Connection connection = getConnection()){
                 // Write the new deliverables to the database and get back the new SERIAL ids
                 // Wrapped in transaction to ensure rollback on bad case_id
                 getContext(connection).transaction(configuration -> {
-                    InsertSetStep deliverableInsertSetStep = PostgresDSL.using(configuration).insertInto(DELIVERABLE_FILE),
-                            deliverableCaseInsertSetStep = PostgresDSL.using(configuration).insertInto(DELIVERABLE_CASE);
-                    InsertValuesStepN deliverableInsertValuesStepN = null,
-                            deliverableCaseInsertValuesStepN = null;
-                    for(Object obj: unknownDeliverables){
+                    // Format JSONArray into a List of Rows so it can be passed to .valuesOfRows
+                    List<Row4<String, String, String, LocalDateTime>> formattedUnknownDeliverables = new ArrayList<>();
+                    for(Object obj: unknownDeliverables) {
                         JSONObject nextDeliverable = (JSONObject) obj;
-                        deliverableInsertValuesStepN = deliverableInsertSetStep.values(
-                                PostgresDSL.defaultValue(), // ID. DEFAULT can't be used in an expression, only as a replacement for an expression. The other not-nulls will still kill bad requests
-                                checkWritePermission(nextDeliverable.get("project_id"), username),
-                                checkWritePermission(nextDeliverable.get("location"), username),
-                                checkWritePermission(nextDeliverable.get("notes"), username),
-                                checkWritePermission(PostgresDSL.localDateTime(nextDeliverable.get("expiry_date").toString()), username)
-                        );
-
+                        Row4<String, String, String, LocalDateTime> row = DSL.row(nextDeliverable.get("project_id").toString(), nextDeliverable.get("location").toString(), nextDeliverable.get("notes").toString(), LocalDateTime.parse(nextDeliverable.get("expiry_date").toString(), ServiceUtils.DATE_TIME_FORMATTER));
```

Comment:
Is there any risk in using LocalDateTime.parse rather than PostgresDSL.localDateTime ?

Reply:
Also is dropping all those uses of checkWritePermission here safe?

Reply:
This will be locked down to users with ADMINISTRATOR permissions so I think it's safe to drop the check but happy to hear your thoughts on this.

Reply:
I think `checkWritePermissions` was born of concern that even if the front end is locked out, the server could still be `curl`ed to?